tools/analyzer_plugin/lib/src/util/boilerplate_utils.dart

```diff
@@ -47,6 +47,14 @@ void addOverReactGeneratedPartDirective(DartFileEditBuilder builder, Compilation
     lastDirective = unit.directives.last;
   }
 
+  if (lastDirective is PartDirective && overReactGeneratedPartDirectiveIsValid(lastDirective, fileUri)) return;
+
+  final overReactPartDirective = getOverReactGeneratedPartDirective(unit);
+  if(overReactPartDirective != null && !overReactGeneratedPartDirectiveIsValid(overReactPartDirective, fileUri)) {
+    fixOverReactGeneratedPartDirective(builder, unit, fileUri);
```

Comment:
Previously, if an over_react part directive existed, but didn't match the file name, this utility would add a valid over_react part directive without removing the old one. It seemed better to replace the old one.

Reply:
Good call